corehq/apps/ota/tests/test_app_aware_restore.py

```diff
@@ -82,45 +83,60 @@ def tearDownClass(cls):
         domain.delete()
         super(AppAwareSyncTests, cls).tearDownClass()
 
-    def _get_fixture(self, fixtures, fixture_id):
-        matches = [f for f in fixtures if f.attrib.get('id') == fixture_id]
-        if matches:
-            return matches[0]
+    def _get_fixtures_by_prefix(self, fixtures, fixture_id_prefix):
+        return [f for f in fixtures if f.attrib.get('id').startswith(fixture_id_prefix)]
+
+    def _expected_v2_report_fixtures_for_id(self, report_id):
+        return {
+            f'{ReportFixturesProviderV2.id}:index',
+            f'{ReportFixturesProviderV2.id}:{report_id}',
+            f'{ReportFixturesProviderV2.id}-filters:{report_id}',
+        }
 
     def test_report_fixtures_provider_without_app(self):
         """
         ReportFixturesProvider should iterate all apps if app not given
         """
-        from corehq.apps.userreports.reports.data_source import ConfigurableReportDataSource
+        from corehq.apps.userreports.reports.data_source import (
+            ConfigurableReportDataSource,
+        )
         with patch.object(ConfigurableReportDataSource, 'get_data') as get_data_mock:
             get_data_mock.return_value = self.rows
             with mock_datasource_config():
                 fixtures = call_fixture_generator(report_fixture_generator, self.user)
-                reports = self._get_fixture(fixtures, ReportFixturesProviderV1.id).findall('.//report')
-        self.assertEqual(len(reports), 2)
-        report_ids = {r.attrib.get('id') for r in reports}
-        self.assertEqual(report_ids, {'123456', 'abcdef'})
+                v2_report_fixtures = self._get_fixtures_by_prefix(fixtures, ReportFixturesProviderV2.id)
+        self.assertEqual(len(v2_report_fixtures), 5)
+        report_ids = {r.attrib.get('id') for r in v2_report_fixtures}
+        self.assertEqual(
+            report_ids,
+            self._expected_v2_report_fixtures_for_id('123456') | self._expected_v2_report_fixtures_for_id('abcdef')
+        )
 
     def test_report_fixtures_provider_with_app(self):
         """
         ReportFixturesProvider should not iterate all apps if app given
         """
-        from corehq.apps.userreports.reports.data_source import ConfigurableReportDataSource
+        from corehq.apps.userreports.reports.data_source import (
```

Comment:
Nit: Does this throw a circular import exception if you try and move it to the top of the file? If not, it would be great to move this since it's being imported in a lot of test functions.

Reply:
Good catch. Addressed in https://github.com/dimagi/commcare-hq/pull/35179/commits/be79d9715a7be0e523ae6a51f44825082d976b52